fe/fe-core/src/main/java/com/starrocks/external/hive/events/MetastoreEventsProcessor.java

```diff
@@ -0,0 +1,409 @@
+            String resource = entry.getKey();
+            Collection<TableName> tableNames = entry.getValue();
+            for (TableName tableName : tableNames) {
+                String dbName = tableName.getDbName();
+                String tblName = tableName.getTblName();
+                HiveTable table = getHiveTable(resource, dbName, tblName);
+                if (table == null) {
+                    continue;
+                }
+                try {
+                    table.refreshTableColumnStats();
+                } catch (Exception e) {
+                    LOG.warn("Failed to refresh table column statistic on [resource: {}, db: {}, table: {}].",
+                            resource, dbName, table, e);
+                } finally {
+                    refreshColumnsTables.remove(resource, tableName);
+                }
+            }
+        }
+    }
+
+    /**
+     * Process the given list of notification events. Useful for tests which provide a list of events
+     */
+    private void processEvents(List<NotificationEvent> events, String resourceName) {
+        List<MetastoreEvent> filteredEvents = metastoreEventFactory.getFilteredEvents(events, resourceName);
+
+        if (filteredEvents.isEmpty()) {
+            lastSyncedEventIds.put(resourceName, events.get(events.size() - 1).getEventId());
+            return;
+        }
+
+        if (Config.enable_hms_parallel_process_evens) {
+            doExecuteWithPartialProgress(filteredEvents);
+        } else {
+            doExecute(filteredEvents, resourceName);
+        }
+
+        events.forEach(event -> prepareRefreshHiveColumnStats(resourceName, event));
+        lastSyncedEventIds.put(resourceName, events.get(events.size() - 1).getEventId());
+    }
+
+    @Override
+    protected void runAfterCatalogReady() {
+        List<String> resource = Lists.newArrayList(tables.rowKeySet());
+        for (String resourceName : resource) {
+            eventProcessorLock.writeLock().lock();
+            List<NotificationEvent> events = Collections.emptyList();
+            try {
+                events = getNextHMSEvents(resourceName);
+                if (!events.isEmpty()) {
+                    processEvents(events, resourceName);
+                }
+            } catch (MetastoreNotificationFetchException e) {
+                LOG.error("Failed to fetch hms events on {}. msg: ", resourceName, e);
+            } catch (Exception ex) {
+                LOG.error("Failed to process hive metastore [{}] events " +
+                        "in the range of event id from {} to {}.", resourceName,
+                        events.get(0).getEventId(), events.get(events.size() - 1).getEventId(), ex);
+            } finally {
+                eventProcessorLock.writeLock().unlock();
+            }
+        }
+    }
+
+    public static TableName getTablesColumnKey(String dbName, String tableName) {
```

Comment:
What does column in the function name refer?

Reply:
This function name is really easy to be a little redundant. it's just to create an TableName instance.